source-code/core/messages/src/query.test.ts

```diff
@@ -0,0 +1,115 @@
+import { createQuery } from "./query.js"
+import { it, describe, expect } from "vitest"
+import type { Message } from "./schema.js"
+
+// TODO: test return value of mutation functions
+
+describe("create", () => {
+	const query = createQuery([])
+
+	it("should create a message", () => {
+		expect(query.get({ where: { id: "first-message" } })).toBeUndefined()
+
+		const mockMessage = createMessage({ id: "first-message", text: "Hello World" })
+
+		query.create({ data: mockMessage })
+
+		expect(query.get({ where: { id: "first-message" } })).toEqual(mockMessage)
+	})
+})
+
+describe("get", () => {
+	const query = createQuery([])
+
+	it("should return undefined if a message does not exist", () => {
+		const message = query.get({ where: { id: "none-existent-message" } })
+
+		expect(typeof message).toBe("undefined")
+	})
+
+	// we assume that each message has a unique id
+	// hence, querying by id should return a single message
+	it("should return an object, not an array", () => {
+		query.create({ data: createMessage({ id: "first-message", text: "Hello World" }) })
+
+		const message = query.get({ where: { id: "first-message" } })
+
+		expect(typeof message).toBe("object")
+	})
+})
+
+describe("getAll", () => {
+	it("should return an empty array if no messages exist", () => {
+		const query = createQuery([])
+		const messages = query.getAll()
+
+		expect(messages).toEqual([])
+	})
+	it("should return all message objects", () => {
+		const query = createQuery([])
+		const mockMessage1 = createMessage({ id: "first-message", text: "Hello World" })
+		const mockMessage2 = createMessage({ id: "second-message", text: "Hello World 2" })
+
+		query.create({ data: mockMessage1 })
+		query.create({ data: mockMessage2 })
+
+		const messages = query.getAll()
+
+		expect(messages).toEqual([mockMessage1, mockMessage2])
+	})
+})
+
+describe("update", () => {
```

Comment:
@NiklasBuchfink what do you actually test here?
In the first test you check if the `get` after `update` returnes the passed object, but the description says something different.
In the other two tests you never call `update`.

Reply:
@ivanhofer I got something mixed up there, but you have already corrected it. Thank you for pointing it out to me.